components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/dao/SAMLSSOServiceProviderDAO.java

```diff
@@ -517,7 +522,12 @@ public boolean updateServiceProvider(SAMLSSOServiceProviderDO serviceProviderDO)
             if (!isTransactionStarted) {
                 registry.beginTransaction();
             }
-            registry.put(path, resource);
+            // Delete the current resource if the issuer value is updated.
+            if (isIssuerUpdated) {
```

Comment:
Shouldn't we create the new resource before deleting the old one? Can you validate the behaviour with the old implementation?

Reply:
In the old implementation. Delete method is called before the add method.

Reply:
With the current implementation, both delete and add functions happens inside a transaction therefore it won't be an issue with the deletion of resource